docs/src/modules/java/pages/workflows.adoc

```diff
@@ -134,6 +134,23 @@ IMPORTANT: We are returning the internal state directly back to the requester. I
 
 A full transfer workflow source code sample can be downloaded as a link:../java/_attachments/workflow-quickstart.zip[zip file]. Follow the `README` file to run and test it.
 
+== Deleting state
+
+If you want to delete the workflow state, you can use the `effect().delete` method. This will remove the workflow from the system.
+
+[source,java,indent=0]
+.{sample-base-url}/transfer-workflow/src/main/java/com/example/transfer/application/TransferWorkflow.java[TransferWorkflow.java]
+----
+include::example$transfer-workflow/src/main/java/com/example/transfer/application/TransferWorkflow.java[tag=delete-workflow]
+----
+<1> Instruction to delete the workflow.
+
+When you give the instruction to delete a running workflow it's equivalent to ending and deleting a workflow. For already finished workflows, it is possible to delete them in the command handler, but any state changes will be ignored. The actual removal of workflow state is delayed to give downstream consumers time to process all prior updates. Including the fact that the workflow has been deleted (via method annotated with `@DeleteHandler`). By default, the existence of the workflow is completely cleaned up after a week.
```

Comment:
are state changes ignored or will it fail with an exception?

Reply:
ignored

Reply:
I think we discussed that with @octonato, whether to crash or not, the reasoning was that for some circumstances a code like:
```
effects()
      .updateState(...)
      .delete()
      .thenReply(done());
```
will update the state, end the workflow, and mark for deletion (for a running workflow). It will ignore the update for an already finished workflow, which, I agree, is not clear. If we decide to crash, then the correct user code should be:
```
public Effect<Done> updateAndDelete(Transfer transfer) {
    if (isDeleted()){
      return effects().reply(...)
    } else {
      return effects()
        .updateState(...)
        .delete()
        .thenReply(done());
    }

```

Maybe that's fine. @octonato WDYT?

Reply:
Wrong sample, now I remember, it must be sth like:
```
public Effect<Done> updateAndDelete(Transfer transfer) {
    if (state.isFinished()){
      return effects()
        .delete()
        .thenReply(done());
    } else {
      return effects()
        .updateState(...)
        .delete()
        .thenReply(done());
    }
```
So the user is responsible for keeping the info about whether the workflow has finished or not, we don't expose that info.

Reply:
The problem with crashing is that differently to ESE and KVE, an Workflow is an autonomous entity. 

For entities, you have an incoming command hitting a command handler that fails.
While a workflow, this can be the `andThen` of a step and then, depending how the step is configured, it's possible it will retry it and crash forever requiring human intervention and possibly a redeploy if they don't have a method that interrupts the workflow.

We can make a distinction from an effect coming a command handler and one coming from an `andThen` and have different behavior for it. But having different behavior is also not great.

Reply:
this particular issue is not a problem for `andThen` because we know that, the workflow is still running and we can update the state and delete it, for the command handler after some discussion with @patriknw I will throw an exception for cases when you want to update the state of finished workflow, this might be a user bug so it's better to be explicit about it, runtime change: https://github.com/lightbend/kalix-runtime/pull/3842

Reply:
docs updated